### PR TITLE
A few relevant papers which have landed in (or near) the Hotspot VM.

### PR DESCRIPTION
**Pull request by bitbckt**

### CODE REVIEW

File: java.md [hunk 1]
```diff
@@ -231,6 +231,46 @@ rounding modes specified by the IEEE-754 standard. They will also
 allow one to easily test libraries that are claimed to provide
 correctly rounded functions.
 
+### [Optimizing Dynamically-Typed Object-Oriented Languages With Polymorphic Inline Caches](http://www.cs.ucsb.edu/~urs/oocsb/papers/ecoop91.pdf)
+
+_Urs Hölzle, Craig Chambers and David Ungar_
```
Comment (evnm) [inline java.md]: Oxford comma, please

File: java.md [hunk 2]
```diff
@@ -231,6 +231,46 @@ rounding modes specified by the IEEE-754 standard. They will also
...
+languages have been limited by the paucity of type information
+available to the compiler. The abundance of the type information
+provided by PICs suggests a new compilation approach for these
+languages, adaptive compilation. Such compilers may succeed in
+generating very efficient code for the time-critical parts of a
+program without incurring distracting compilation pauses.
+
+### [Optimizing Dynamically-Dispatched Calls with Run-Time Type Feedback](http://www.cs.ucsb.edu/~urs/oocsb/papers/pldi94.pdf)
+
+_Urs Hölzle, and David Ungar_
```
Comment (evnm) [inline java.md]: Rm oxford comma

File: java.md [hunk 3]
```diff
@@ -307,3 +347,124 @@ outperform the Java SE 5.0 SynchronousQueue by factors of three in
 unfair mode and 14 in fair mode; this translates to factors of two and
 ten for the ThreadPoolExecutor. Our synchronous queues have been
 adopted for inclusion in Java 6.
+
+### [A Simple Graph-Based Intermediate Representation](http://www.oracle.com/technetwork/java/javase/tech/c2-ir95-150110.pdf)
+
+_Cliff Click, and Michael Paleczny_
```
Comment (evnm) [inline java.md]: ditto

File: java.md [hunk 4]
```diff
@@ -307,3 +347,124 @@ outperform the Java SE 5.0 SynchronousQueue by factors of three in
...
+We represent the CFG and basic blocks with the same
+vertex and edge structures. Each opcode is defined by a
+C++ class that encapsulates opcode-specific data and be-
+havior. We use inheritance to abstract common opcode
+behavior, allowing new opcodes to be easily defined from
+old ones. The resulting IR is simple, fast and easy to use.
+
+### [Linear Scan Register Allocation on SSA Form](http://www.christianwimmer.at/Publications/Wimmer10a/Wimmer10a.pdf)
+
+_Christian Wimmer, and Michael Franz_
```
Comment (evnm) [inline java.md]: ditto

File: java.md [hunk 5]
```diff
@@ -307,3 +347,124 @@ outperform the Java SE 5.0 SynchronousQueue by factors of three in
...
+the work of tracing the object graph, using a
+form of dynamic load balancing called work stealing.
+We describe two collectors written using this
+framework: pSemispaces, a parallel semispace collector,
+and pMarkcompact, a parallel markcompact
+collector.
+
+### [A Generational Mostly-concurrent Garbage Collector](http://www.cs.ucsb.edu/~ckrintz/racelab/gc/papers/detlefs-generational.pdf)
+
+_Tony Printezis, and David Detlefs_
```
Comment (evnm) [inline java.md]: ditto

**Review by evnm — CHANGES_REQUESTED** [review]
Dumb nits

**Comment by AlexDenisov** [comment]
Aren't papers supposed to be in alphabetical order?

**Comment by evnm** [comment]
Good point. The Java and .NET lists are in the same order in which they appeared on your and @mattwarren's blogs. Which of the following schemes do you think it'd make most sense to standardize on?

- Alphabetical by title
- Alphabetical by first author
- Chronological by date of publication

**Comment by AlexDenisov** [comment]
I used alphabetical order by title.
Also, I realised that so far only my list has some particular order.
So it's totally up to you to decide, @evnm :)

**Comment by bitbckt** [comment]
Sort order wasn't defined in the guidelines. I'm happy to comply with whichever regime you prefer.

**Comment by evnm** [comment]
I think we should order publications alphabetically by first author, since this is what most IEEE-style papers themselves do. However, let's tackle that in a separate change.

**Review by evnm — APPROVED** [review]

**Pull request merged**